### ISSUE TITLE
Nullable Typehint is not detected as nullable

### ISSUE THREAD

**Issue by wendelb** [squizlabs/PHP_CodeSniffer#3782]
**Describe the bug**

**Code sample**
```php
class MyClass {
    private function myFunction(ClassA|ClassB|null $param): bool {
    }
}
```

**To reproduce**
1. Acquire the method parameters in a Sniff that is registered for functions (`\PHP_CodeSniffer\Files\File::getMethodParameters`)
2. Test if the only parameter is nullable
```

**Actual behavior**
The Type Hint is detected as `ClassA|ClassB|null`, but the property `"nullable_type"` is set to false.

**Expected behavior**
PHPCS should be able to detect that the $param parameter is nullable

**Versions (please complete the following information):**
 - OS: N/A (Ubuntu, some current version)
 - PHP: 8.1
 - PHPCS: 3.7.1
 - Standard: N/A

**Additional context**
The Sniff here https://github.com/spryker/code-sniffer/blob/master/Spryker/Sniffs/Commenting/DocBlockParamAllowDefaultValueSniff.php is using get `getMethodParameters` method to validate docblock and typehint. Due to wrong input from PHPCS it create a violation where none should be.

**Comment by jrfnl** [comment]
This is intentional and was discussed in detail when support for union types was implemented. Please read the discussion here: https://github.com/squizlabs/PHP_CodeSniffer/issues/2968#issuecomment-653948953

**Comment by gsherwood** [comment]
Hopefully that linked comment from @jrfnl helped explain the behaviour.
